### ISSUE TITLE
typescript feature

### ISSUE THREAD

**Issue by JeffGreat** [diegohaz/rest#228]
Hi, is there a typescript version of the repo. I'm using this boilerplate since years and I'm very satisfied with it. Now the project might need some refresh to include new features that can be brought with typescript and other libs.
Does anybody intersting by improving this repo ?
Thx and special thx to Diego :)

**Comment by fuadnafiz98** [comment]
I am interested to migrate this project in typescript, I am exploring this repo. What's your thoughts?

**Comment by JeffGreat** [comment]
Actually I would like to get a better insight of alternative of this repo. Does it worth to migrate to TS ? 
Lastly we've implemented Nestjs for one project. Looks interested even if it's quite heavy as a framework can be. Another thought, is express still valuable for this kind of rest api task ?

**Comment by fuadnafiz98** [comment]
yah, I learned Nest.js and it is in fact quite heavy, but it is production-ready and scalable, but I think of trying out restify, it's faster than express as far as I know. also, nestjs requires quite a lot of work to build an API.

**Comment by JeffGreat** [comment]
I do agree about nestjs. I've checked out restify and it's pretty much like express. It's also very similar to the way @diegohaz implements rest in this project.
Have you already check Loopback framework ? There are also those ones https://expressjs.com/en/resources/frameworks.html

**Comment by fuadnafiz98** [comment]
No, I never used the Loopback framework, Do you think Loopback is a good choice? I am open to it :)

**Comment by JeffGreat** [comment]
I've checked quickly FoalTs yesterday. https://github.com/FoalTS/foal. It looks quite fine. It seems to be more or less like nestJs but a lighter. They had the good idea to publish framework comparison:
https://foalts.gitbook.io/docs/topic-guides/comparison-with-other-frameworks

According to them, loopback might be too light for complex projects. Nestjs is powerfull and has traction but it is quite opinionated as its structure is inspired by angular.

In conclusion, I'll continue to use nestjs and I'll try FoalTs in a future project in order to have my opinion.

**Comment by bhanuc** [comment]
I am planning to migrate this project to typescript, as I want to use a similar template written for typescript for my future projects. @diegohaz will you be interested adding typescript support in the project ?
I have understood the basic functionality of this generator and can help with a pr as well.
